presto-main/src/main/java/com/facebook/presto/memory/QueryContext.java

```diff
@@ -129,13 +131,20 @@ public QueryContext(
                 newRootAggregatedMemoryContext(new QueryMemoryReservationHandler(this::updateSystemMemory, this::tryReserveMemoryNotSupported, this::updateBroadcastMemory, this::tryUpdateBroadcastMemory), 0L));
     }
 
+    public boolean needsMemoryLimitsInitialized()
+    {
+        return needsMemoryLimitsInitialized;
+    }
+
     // TODO: This method should be removed, and the correct limit set in the constructor. However, due to the way QueryContext is constructed the memory limit is not known in advance
     public synchronized void setResourceOvercommit()
     {
         // Allow the query to use the entire pool. This way the worker will kill the query, if it uses the entire local general pool.
         // The coordinator will kill the query if the cluster runs out of memory.
         maxUserMemory = memoryPool.getMaxBytes();
```

Comment:
It's unclear to me if it's intentional (or desirable) that a task update would previously cause these values to change once a query was promoted to the reserved pool. If so, it's easy enough to modify this PR to handle that as well- just need to know the intended behavior in that situation.

Reply:
Since it's unclear what the intended or correct interaction between resource over-commit and the reserved pool should be in this scenario, I've opted to preserve the current effective behavior by modifying the memory limits for resource over-commit queries as part of their memory pool change.

Reply:
This change makes sense to me. The limit on resource over commit should be all available memory in the pool. We don't want to fail without using the entire pool. If the reserve pool happens to be larger then we want to use it.

I don't see that as being common but whatever :-)